### PR TITLE
LibJS: Add the FinalizationRegistry built-in object

### PR DESCRIPTION
**Pull request by IdanHo**
This fixes 51 test262 test cases (Note that there are 8 more test cases that look like they should pass, but currently dont due to parser errors related to class declarations).

As part of this PR the FinalizationRegistery.prototype.cleanupSome method was added, which is actually still a stage 2 proposal, but since test262 test cases already exist for it, i decided to go for it :)

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/FinalizationRegistry.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+void FinalizationRegistry::cleanup(Function* callback)
+{
+    auto cleanup_callback = callback ?: m_cleanup_callback;
+    for (auto it = m_records.begin(); it != m_records.end(); ++it) {
+        if (it->target != nullptr)
+            continue;
+        (void)vm().call(*cleanup_callback, js_undefined(), it->held_value);
+        m_records.remove(it);
+        if (vm().exception())
+            return;
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/FinalizationRegistry.cpp]: Let's keep a reference to the VM at the beginning of the function as usual (`auto& vm = this->vm()`) - `vm.foo()` reads a bit nicer later on :)
(and technically slightly more performant, although I trust the compiler to do the right thing)

File: Userland/Libraries/LibJS/Runtime/FinalizationRegistryPrototype.cpp [hunk 2]
```diff
@@ -38,4 +43,68 @@ FinalizationRegistry* FinalizationRegistryPrototype::typed_this(VM& vm, GlobalOb
...
+    if (!finalization_registry)
+        return {};
+
+    auto callback = vm.argument(0);
+    if (vm.argument_count() > 0 && !callback.is_function()) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::NotAFunction, callback.to_string_without_side_effects());
+        return {};
+    }
+
+    finalization_registry->cleanup(&callback.as_function());
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/FinalizationRegistryPrototype.cpp:59]: This will blow up if `callback` is not present, i.e. undefined.

Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/FinalizationRegistryPrototype.cpp:59]:
```suggestion
    finalization_registry->cleanup(callback.is_undefined() ? nullptr : &callback.as_function());
```

**Review by linusg — COMMENTED** [review]
Fantastic work! :tada:

File: Userland/Libraries/LibJS/Runtime/FinalizationRegistry.cpp (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/FinalizationRegistry.cpp]: done :)

File: Userland/Libraries/LibJS/Runtime/FinalizationRegistryPrototype.cpp (hunk 2 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/FinalizationRegistryPrototype.cpp]: my bad, fixed!

**Pull request merged**